src/main/java/io/supertokens/inmemorydb/queries/JWTSigningQueries.java

```diff
@@ -16,20 +16,93 @@
+                + "key_id VARCHAR(255) NOT NULL," + "key_string TEXT NOT NULL,"
+                + "algorithm VARCHAR(10) NOT NULL," + "created_at BIGINT UNSIGNED,"
                 + "PRIMARY KEY(key_id));";
     }
 
-    static String getQueryToFetchLatestKey(Start start) {
-        return "SELECT * FROM " + Config.getConfig(start).getJWTSigningKeysTable()
-                + "ORDER BY created_at DESC";
+    public static List<JWTSigningKeyInfo> getJWTSigningKeys_Transaction(Start start, Connection con)
+            throws SQLException, StorageQueryException {
+
+        ((ConnectionWithLocks) con).lock(lockString);
+
+        String QUERY = "SELECT * FROM "
+                + Config.getConfig(start).getJWTSigningKeysTable()
+                + " ORDER BY created_at DESC;";
+
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            ResultSet result = pst.executeQuery();
+            List<JWTSigningKeyInfo> keys = new ArrayList<>();
+
+            while(result.next()) {
+                keys.add(JWTSigningKeyInfoRowMapper.getInstance().mapOrThrow(result));
+            }
+
+            return keys;
+        }
+    }
+
+    private static class JWTSigningKeyInfoRowMapper implements RowMapper<JWTSigningKeyInfo, ResultSet> {
+        private static final JWTSigningKeyInfoRowMapper INSTANCE = new JWTSigningKeyInfoRowMapper();
+
+        private JWTSigningKeyInfoRowMapper() {
+        }
+
+        private static JWTSigningKeyInfoRowMapper getInstance() {
+            return INSTANCE;
+        }
+
+        @Override
+        public JWTSigningKeyInfo map(ResultSet result) throws Exception {
+            String keyId = result.getString("key_id");
+            String keyString = result.getString("key_string");
+            long createdAt = result.getLong("created_at");
+            String algorithm = result.getString("algorithm");
+
+            if (keyString.contains("|")) {
+                return JWTAsymmetricSigningKeyInfo.withKeyString(keyId, createdAt, algorithm, keyString);
+            } else {
+                return new JWTSymmetricSigningKeyInfo(keyId, createdAt, algorithm, keyString);
+            }
+        }
+    }
+
+    public static void setJWTSigningKeyInfo_Transaction(Start start, Connection con, JWTSigningKeyInfo info)
+            throws StorageTransactionLogicException {
+
+        String QUERY = "INSERT INTO " + Config.getConfig(start).getJWTSigningKeysTable()
+                + "(key_id, key_string, created_at, algorithm) VALUES(?, ?, ?, ?)";
+
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            pst.setString(1, info.keyId);
+            pst.setString(2, info.keyString);
+            pst.setLong(3, info.createdAtTime);
+            pst.setString(4, info.algorithm);
+            pst.executeUpdate();
+        } catch (SQLException e) {
+            throw new StorageTransactionLogicException(e);
+        }
```

Comment:
Why are you not throwing the `SQLException` directly?

Reply:
Done